File: RestaurantOrdersWebApplication/src/app/models/orderdetails/OrderDetailsViewModel.ts
import { OrderViewModel } from "../order/OrderViewModel";
import { ProductViewModel } from "../product/ProductViewModel";

export class OrderDetailsViewModel {
    id: number;
    orderId: number;
    productId: number;
    quantity: number;
    productName: string;
    price: number;
    totalOrderPrice: number;

    order: OrderViewModel;
    product: ProductViewModel;
}

File: RestaurantOrdersApplicationService/RestaurantOrdersApplicationService.Api/ViewModels/OrderDetails/OrderDetailsViewModel.cs
using RestaurantOrdersApplicationService.Api.ViewModels.Order;
using RestaurantOrdersApplicationService.Api.ViewModels.Product;

namespace RestaurantOrdersApplicationService.Api.ViewModels.OrderDetails
{
    public class OrderDetailsViewModel
    {
        public int Id { get; set; }
        public int OrderId { get; set; }
        public int ProductId { get; set; }
        public int Quantity { get; set; }

        public OrderViewModel Order { get; set; }
        public ProductViewModel Product { get; set; }
    }
}
